back/engines/free/email_campaigns/app/services/email_campaigns/delivery_service.rb

```diff
@@ -171,6 +165,21 @@ def send_command_internal(campaign, command)
         .campaign_mail
         .deliver_later(wait: command[:delay] || 0)
     end
+
+    def generate_commands(campaign, recipient, options = {})
+      campaign.generate_commands(recipient:, **options).map do |command|
+        command.merge(
+          recipient: recipient,
+          time: Time.zone.now,
+          delivery: campaign.deliveries.new( # Needed to include the delivery ID in the Maigun headers
```

Comment:
Alternatively, perhaps I should have just added `delivery_id: SecureRandom.uuid` and save the delivery in the trackable concern more similarly as before. I might make that change tomorrow.